projects/BFDR/BirthRecord_submissionProperties.cs

```diff
@@ -3521,5 +3519,275 @@ public string FatherEthnicityLiteral
+        {
+            // TODO add validation of value once ValueSets.cs has been generated.
+            var entry = Bundle.Entry.Where(e => e.Resource is Observation o && CodeableConceptToDict(o.Code)["code"] == code).FirstOrDefault();
+            if (!(entry?.Resource is Observation obs))
+            {
+                obs = SetWeight(code, null, "", section, subjectId);
+            }
+
+            // If there's a value clear this extension in case it's previously set, otherwise set an empty value
+            if (obs.Value == null)
+            {
+                obs.Value = new CodeableConcept();
+            }
+            else
+            {
+                obs.Value.Extension.RemoveAll(ext => ext.Url == VRExtensionURLs.BypassEditFlag);
+            }
+            Extension extension = new Extension(VRExtensionURLs.BypassEditFlag, DictToCodeableConcept(value));
+            obs.Value.Extension.Add(extension);
+        }
+
+        private void SetWeightEditFlagHelper(string code, string editFlag, string section, string subjectId)
+        {
+            // TODO add validation of editFlag and automate code system extraction once ValueSets.cs is available
+            if (String.IsNullOrEmpty(editFlag))
+            {
+                SetWeightEditFlag(code, EmptyCodeDict(), section, subjectId);
+                return;
+            }
+            Dictionary<string, string> dictionary = new Dictionary<string, string>
+            {
+                { "code", editFlag },
+                { "system", "http://hl7.org/fhir/us/vr-common-library/CodeSystem/CodeSystem-vr-edit-flags" }
+            };
+            SetWeightEditFlag(code, dictionary, section, subjectId);
+        }
+
+        /// <summary>Mother's Prepregnancy Weight Edit Flag.</summary>
+        /// <value>edit flag for the mother's prepregnancy weight</value>
+        /// <example>
+        /// <para>// Setter:</para>
+        /// <para>Dictionary&lt;string, string&gt; edit = new Dictionary&lt;string, string&gt;();</para>
+        /// <para>edit.Add("code", "0");</para>
+        /// <para>edit.Add("system", "http://hl7.org/fhir/us/vr-common-library/CodeSystem/CodeSystem-vr-edit-flags");</para>
+        /// <para>edit.Add("display", "Edit Passed");</para>
+        /// <para>ExampleBirthRecord.MotherPrepregnancyWeightEditFlag = route;</para>
+        /// <para>// Getter:</para>
+        /// <para>Console.WriteLine($"Mother's prepregnancy weight edit flag: {ExampleBirthRecord.MotherPrepregnancyWeightEditFlag}");</para>
+        /// </example>
+        [Property("MotherPrepregnancyWeightEditFlag", Property.Types.Dictionary, "Mother Prenatal", "Mother Prenatal, Weight at Delivery Edit Flag", true, IGURL.ObservationMotherPrepregnancyWeight, true, 138)]
+        [FHIRPath("Bundle.entry.resource.where($this is Observation).where(code.coding.code='56077-1')", "")]
+        public Dictionary<string, string> MotherPrepregnancyWeightEditFlag
```

Comment:
Since the helper doesn't actually use this, do we need it?

Reply:
If we only care about whether the helper uses it, then no but I thought we wanted the "real" coded value in the FHIR API and that the helper was just a convenience method for use in the IJE converter?